packages/grafana-ui/src/components/Combobox/MultiCombobox.tsx

```diff
@@ -262,31 +306,29 @@ function getSelectedItemsFromValue<T extends string | number>(
   value: T[] | Array<ComboboxOption<T>>,
   options: Array<ComboboxOption<T>>
 ) {
-  if (!isComboboxOptions(value)) {
-    const resultingItems: Array<ComboboxOption<T> | undefined> = [];
-
-    for (const item of options) {
-      for (const [index, val] of value.entries()) {
-        if (val === item.value) {
-          resultingItems[index] = item;
-        }
-      }
-      if (resultingItems.length === value.length && !resultingItems.includes(undefined)) {
-        // We found all items for the values
-        break;
-      }
-    }
+  if (isComboboxOptions(value)) {
```

Comment:
The changes to this function handles a severe performance issue (nested loops 🙄 )